ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt

```diff
@@ -58,61 +105,190 @@ class MainActivity : ComponentActivity() {
-                ) { }
-            }
-            Column {
-                ButtonList(
-                    onLoadModel = {
-                        mService?.send(
-                            Message.obtain(
-                                null,
-                                MessageType.LOAD_MODELS.value
-                            )
-                        )
+                    color = colorScheme.background
+                ) {
+                    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+                    val scope = rememberCoroutineScope()
+                    var selectedItemIndex by rememberSaveable {
+                        mutableIntStateOf(0)
                     }
-                )
-                ServiceList(
-                    mViewModel.services.collectAsState().value,
-                    onClickStart = { id ->
-                        mService?.send(
-                            Message.obtain(
-                                null,
-                                MessageType.START_MODEL.value,
-                                id,
-                                0
-                            )
-                        )
-                    },
-                    onClickStop = { id ->
-                        mService?.send(
-                            Message.obtain(
-                                null,
-                                MessageType.STOP_MODEL.value,
-                                id,
-                                0
-                            )
-                        )
-                    },
-                    onClickDestroy = { id ->
-                        mService?.send(
-                            Message.obtain(
-                                null,
-                                MessageType.DESTROY_MODEL.value,
-                                id,
-                                0
-                            )
-                        )
-                    })
+
+                    ModalNavigationDrawer(
+                        drawerContent = {
+                            ModalDrawerSheet {
+                                Spacer(modifier = Modifier.height(16.dp))
+                                items.forEachIndexed { index, item ->
+                                    NavigationDrawerItem(
+                                        label = {
+                                            Text(text = item.title)
+                                        },
+                                        selected = index == selectedItemIndex,
+                                        onClick = {
+                                            selectedItemIndex = index
+                                            scope.launch {
+                                                drawerState.close()
+                                            }
+                                            navController.navigate(item.screen)
```

Comment:
```suggestion
                                            navController.navigate(item.screen) {
                                                popUpTo(navController.graph.id) {
                                                    inclusive = true
                                                }
                                            }
```

How about adding inclusive option?
I don't know much about navigation drawer but I found `navController` pushs every composable? to back stack when I clicked option and changes view.
So if I clicked `Vision Examples` -> `Settings` -> `Vision Examples` -> `MLAgent`, I have to press back button 4 times to close the app

I'm not sure pop every elements from back stack is the right way, but suggested code is for poping up every views and closes the application when user pressed back button.

https://developer.android.com/guide/navigation/backstack#compose-examples
https://velog.io/@beokbeok/Jetpack-Compose-%EB%B0%B1%EC%8A%A4%ED%83%9D-%EC%A0%95%EB%A6%AC

Reply:
Thanks @niley7464 !